tests/integration/buildcmd/test_build_cmd.py

```diff
@@ -266,8 +251,61 @@ def _verify_built_artifact(self, build_dir, function_logical_id, expected_files,
 
         self.assertTrue(any([True if self.EXPECTED_RUBY_GEM in gem else False for gem in os.listdir(str(gem_path))]))
 
-    def _verify_resource_property(self, template_path, logical_id, property, expected_value):
 
-        with open(template_path, 'r') as fp:
-            template_dict = yaml_parse(fp.read())
-            self.assertEquals(expected_value, template_dict["Resources"][logical_id]["Properties"][property])
+class TestBuildCommand_GoFunctions(BuildIntegBase):
+
+    EXPECTED_FILES_GLOBAL_MANIFEST = set()
+    EXPECTED_FILES_PROJECT_MANIFEST = {'main.handler'}
+    EXPECTED_BINARY = 'main.handler'
+
+    FUNCTION_LOGICAL_ID = "Function"
+
+    @parameterized.expand([
+        ("go1.x", False),
+        ("go1.x", "use_container")
+    ])
+    def test_with_default_go_mod(self, runtime, use_container):
+        overrides = {"Runtime": runtime, "CodeUri": "Go"}
+        cmdlist = self.get_command_list(use_container=use_container,
+                                        parameter_overrides=overrides,
+                                        # TODO remove this once aws-lambda-builders is released
+                                        skip_pull_image=True)
+
+        LOG.info("Running Command: {}".format(cmdlist))
+        process = subprocess.Popen(cmdlist, cwd=self.working_dir)
+        process.wait()
+
+        self._verify_built_artifact(self.default_build_dir, self.FUNCTION_LOGICAL_ID,
+                                    self.EXPECTED_FILES_PROJECT_MANIFEST)
+
+        self._verify_resource_property(str(self.built_template),
```

Comment:
I don't know if we need to repeat this test for every language.

Reply:
My opinion is yes. This is to help us in not breaking this for a given language. As build grows, we may need to make language specific logic branches, which could lead to the wrong this happening. It's safer to just check it every time, it really doesn't hurt anything and is part of the expected outcome.